lib/util/util.js

```diff
@@ -641,6 +641,71 @@ const Util = {
             Util.logger.debug('CLI logger set to: debug');
         }
     },
+    /**
+     * helper for hasChangedGeneric
+     *
+     * @param {*} oldItem old version of an object
+     * @param {*} newItem new version of an object
+     * @param {string} prop property of object to be checked
+     * @returns {boolean} true if property has changed
+     */
+    isObjectChanged(oldItem, newItem, prop) {
+        if (!oldItem) {
+            return true;
+        }
+        if (
+            newItem[prop] === null ||
+            ['string', 'number', 'boolean'].includes(typeof newItem[prop])
+        ) {
+            // check should ignore types to bypass string/number auto-conversions caused by SFMC-SDK
+            return newItem[prop] != oldItem[prop];
+        } else if (Array.isArray(newItem[prop])) {
+            return this.arrayEquals(oldItem[prop], newItem[prop]);
+        } else {
+            let equal = true;
+            for (const propDeep in newItem[prop]) {
+                equal = equal && this.isObjectChanged(oldItem[prop], newItem[prop], propDeep);
+            }
+            return equal;
+        }
+    },
+    /**
+     * compares to arrays and returns true if they are different
+     *
+     * @param {Array} arrayA first array to compare
+     * @param {Array} arrayB second array to compare
+     * @returns {boolean} true if arrays are equal
+     */
+    arrayEquals(arrayA, arrayB) {
+        // if one of the arrays is a falsy value, return
+        if (!arrayA || !arrayB) {
+            return false;
+        }
+
+        // check if both are actually arrays
+        if (!Array.isArray(arrayA) || !Array.isArray(arrayB)) {
+            return false;
+        }
+
+        // compare lengths - can save a lot of time
+        if (arrayA.length != arrayB.length) {
+            return false;
+        }
+
+        for (let i = 0, l = arrayA.length; i < l; i++) {
+            // Check if we have nested arrays
+            if (Array.isArray(arrayA[i]) || Array.isArray(arrayB[i])) {
+                // recurse into the nested arrays
+                if (!this.arrayEquals(arrayA[i], arrayB[i])) {
+                    return false;
+                }
+            } else if (arrayA[i] != arrayB[i]) {
+                // Warning - two different object instances will never be equal: {x:20} != {x:20}
+                return false;
```

Comment:
if we consider automations, objects would be nested in arrays. So this would force it to always be deployed, when actually that may not be needed.  is that the correct behaviour? https://www.npmjs.com/package/deep-equal would still seem to be a better way to actually check diffs